### ISSUE TITLE
Do automatic solution checking and validation

### ISSUE THREAD

**Issue by whoburg** [convexopt/gpkit#23]
Very low priority:

We could create `GP.solution_check(self, sol)`. It would validate that a particular primal/dual solution pair solves the GP. Could be useful for sanity checks and trying out 'suspect' solution codes.

This is very very very low priority.

**Comment by whoburg** [comment]
To implement -- `gpkit.GP.__run_solver` already does some checks -- can add more there.

**Comment by bqpd** [comment]
Note to do this using the A matrix / matrix math, etc., not substitution.

**Comment by whoburg** [comment]
Will implement this once #226 is complete.

**Comment by bqpd** [comment]
Closing this will close half of the remaining issues! Go @whoburg go

**Comment by whoburg** [comment]
where should this be implemented?

**Comment by bqpd** [comment]
In `geometric_program.solve()`

**Comment by bqpd** [comment]
Probably either just before or just after `self.result = result`: before means that no bad result is made available if these checks fail, after means that the bad result is made available but not returned.

**Comment by whoburg** [comment]
I should be able to work on this in a few hours.

**Comment by whoburg** [comment]
Sorry for delay -- still getting to this, but I want to investigate a possible bug first that I consider higher priority. More info soon.

**Comment by whoburg** [comment]
still psyched to implement this, but waiting until #338 and #335 are fixed.

**Comment by whoburg** [comment]
@bqpd how do I best get the slice of `A` corresponding to:

 - the objective
 - A specific posynomial

?

**Comment by bqpd** [comment]
I think `gp.A.tocsr()[0,:]` will get the objective, and >0 the constraints?

**Comment by whoburg** [comment]
```TypeError: 'coo_matrix' object has no attribute '__getitem__'```

Also, that's just the first monomial in the objective, but I can deal with that via p_idxs

**Comment by whoburg** [comment]
I think I have a solution where I don't actually need the answer to this, but I'm still curious...

**Comment by bqpd** [comment]
Sparse matrices are fun!

(yeah, you need a CSR sparse matrix to index, hence the `tocsr` above)

**Comment by whoburg** [comment]
doh, my fault.

**Comment by whoburg** [comment]
this is coming along -- hard at work
